tests/auto/test_sequential_model.py

```diff
@@ -185,7 +185,59 @@ def test_merge_concat(self):
         nloss = model.evaluate([X_train, X_train], y_train, verbose=0)
         print(nloss)
         assert(loss == nloss)
+    def test_merge_dot(self):
+        print('Test merge: dot')
+        left = Sequential()
+        left.add(Dense(input_dim=input_dim, output_dim=nb_hidden))
+        left.add(Activation('relu'))
+
+        right = Sequential()
+        left.add(Dense(input_dim=input_dim, output_dim=nb_hidden))
+        right.add(Activation('relu'))
+
+        model = Sequential()
+        model.add(Merge([left, right], mode='dot', dot_axes=1))
+
+        model.add(Dense(nb_class))
+        model.add(Activation('softmax'))
+
+        model.compile(loss='categorical_crossentropy', optimizer='rmsprop')
 
+        model.fit([X_train, X_train], y_train, batch_size=batch_size, nb_epoch=nb_epoch, show_accuracy=True, verbose=0, validation_data=([X_test, X_test], y_test))
+        model.fit([X_train, X_train], y_train, batch_size=batch_size, nb_epoch=nb_epoch, show_accuracy=False, verbose=0, validation_data=([X_test, X_test], y_test))
+        model.fit([X_train, X_train], y_train, batch_size=batch_size, nb_epoch=nb_epoch, show_accuracy=True, verbose=0, validation_split=0.1)
+        model.fit([X_train, X_train], y_train, batch_size=batch_size, nb_epoch=nb_epoch, show_accuracy=False, verbose=0, validation_split=0.1)
+        model.fit([X_train, X_train], y_train, batch_size=batch_size, nb_epoch=nb_epoch, verbose=0)
+        model.fit([X_train, X_train], y_train, batch_size=batch_size, nb_epoch=nb_epoch, verbose=0, shuffle=False)
+
+        loss = model.evaluate([X_train, X_train], y_train, verbose=0)
+        print('loss:', loss)
+        if loss > 0.7:
+            raise Exception('Score too low, learning issue.')
+        preds = model.predict([X_test, X_test], verbose=0)
+        classes = model.predict_classes([X_test, X_test], verbose=0)
+        probas = model.predict_proba([X_test, X_test], verbose=0)
+        print(model.get_config(verbose=1))
```

Comment:
Everything below isn't necessary since these features do not depend on the merge mode and are being tested in a different test method.